MyCryptoHQ/MyCrypto
# Electron Redesign

Not tracked as an issue.

### Description

This PR aims to make our electron app feel more like just that, an app. It moves the navigation off to the side, and gets rid of the header and footer. There was also a bit of refactoring that went into this to make certain components more agnostic about how they're used.

The PR looks a lot bigger than it really is, everything here is just styling and component changes, no changes to redux / libs / utils.

### Changes

* Electron app now has a dope sidebar that handles navigation, node switching, and language switching.
* OSX app no longer has a frame
    * This was a little too tricky to implement and test on Windows and Linux in the short time I had, but I plan to revisit those platforms at a later date.
* Added a "Support Us" page only on the electron app that has all of the information in the footer.
* Modals now use [React portals](https://reactjs.org/docs/portals.html), so that they always render correctly
    * This allows elements that have opacity or overflow hidden to render modals with no problem. This also makes it so modals are unaffected by the styling of the component they're contained in.
* `TabSection` now uses either `WebTemplate` or `ElectronTemplate` depending on environment
* Navigation links and footer links have been moved into `config/navigation.ts` and `config/links.ts` for reusability
* Exposes `$is-electron` variable to sass.
    * This is primarily to make breakpoints happen at larger sizes, due to the sidebar
* Added an external link icon to "Help" in the navigation

### Steps to Test

1. Run the electron app in dev, provide feedback
2. Run the web app, make sure nothing changed
3. Try to build the electron app for your platform, make sure it's not busted

Some areas that I'm either concerned about, or would like feedback on:

* Not wild about the language / network buttons on the frame, but couldn't come up with anything better
* Modals currently cover everything, including the nav. Does that look weird?
* The navigation text is quite small to keep footprint low, does this render alright on all operating systems?

## Screenshots

### OSX, Homepage

![screen shot 2018-04-12 at 5 46 00 pm](https://user-images.githubusercontent.com/649992/38705951-630898fa-3e79-11e8-98e4-78b75d1b8766.png)

### OSX, Swap, Changing Networks

![screen shot 2018-04-12 at 5 48 37 pm](https://user-images.githubusercontent.com/649992/38706048-c0380a06-3e79-11e8-81cc-a6f05d830038.png)

More platform screenshots forthcoming
